File: leetcode/no1598_Crawler_Log_Folder.go
package main

import "fmt"

func minOperations(logs []string) int {
	level := 0

	for _, log := range logs {
		switch log {
		case "./":
			level = level
		case "../":
			level -= 1
			if level < 0 {
				level = 0
			}
		default:
			level += 1
		}
	}

	return level
}

func main() {

	logs := []string{"d1/", "d2/", "./", "d3/", "../", "d31/"}
	result := minOperations(logs)
	fmt.Println("OUTPUT:", result)
}

File: leetcode/no1598_Crawler_Log_Folder.js
/**
 * @param {string[]} logs
 * @return {number}
 */
var minOperations = function (logs) {
    let level = 0;

    logs.forEach((log) => {
        if (log == "../") {
            level -= 1;
            if (level < 0) {
                level = 0;
            }
        } else if (log == "./") {
            // don't move
        } else {
            level += 1;
        }
    })

    return level;
};


let logs = ["d1/", "d2/", "../", "d21/", "./"]
minOperations(logs);